diff --git a/python/mozversioncontrol/mozversioncontrol/__init__.py b/python/mozversioncontrol/mozversioncontrol/__init__.py
--- a/python/mozversioncontrol/mozversioncontrol/__init__.py
+++ b/python/mozversioncontrol/mozversioncontrol/__init__.py
@@ -957,14 +957,14 @@
 
     def get_upstream(self):
         pass
 
     def get_changed_files(self, diff_filter="ADM", mode="unstaged", rev=None):
-        pass
+        return []
 
     def get_outgoing_files(self, diff_filter="ADM", upstream=None):
-        pass
+        return []
 
     def add_remove_files(self, *paths: Union[str, Path]):
         pass
 
     def forget_add_remove_files(self, *paths: Union[str, Path]):

