# Backport of https://github.com/mozilla/authenticator-rs/pull/246
--- a/third_party/rust/authenticator/src/transport/windows/winapi.rs
+++ b/third_party/rust/authenticator/src/transport/windows/winapi.rs
@@ -221,7 +221,7 @@ impl DeviceInterfaceDetailData {
     }
 
     fn path(&self) -> String {
-        unsafe { from_wide_ptr((*self.data).DevicePath.as_ptr(), self.path_len - 2) }
+        unsafe { from_wide_ptr(ptr::addr_of!((*self.data).DevicePath[0]), self.path_len - 2) }
     }
 }
 
