diff --git a/browser/base/content/browser.js b/browser/base/content/browser.js
index b63e0cd..4e3e53b 100644
--- a/browser/base/content/browser.js
+++ b/browser/base/content/browser.js
@@ -5504,8 +5504,6 @@ var XULBrowserWindow = {
 
     SafeBrowsingNotificationBox.onLocationChange(aLocationURI);
 
-    SaveToPocket.onLocationChange(window);
-
     UrlbarProviderSearchTips.onLocationChange(
       window,
       aLocationURI,
diff --git a/browser/components/BrowserGlue.jsm b/browser/components/BrowserGlue.jsm
index 43fbae6..5db0911 100644
--- a/browser/components/BrowserGlue.jsm
+++ b/browser/components/BrowserGlue.jsm
@@ -1270,7 +1270,6 @@ BrowserGlue.prototype = {
       lazy.Normandy.init();
     }
 
-    lazy.SaveToPocket.init();
 
     AboutHomeStartupCache.init();
 
diff --git a/browser/components/moz.build b/browser/components/moz.build
index 3b2126e..32b7831 100644
--- a/browser/components/moz.build
+++ b/browser/components/moz.build
@@ -44,7 +44,6 @@ DIRS += [
     "originattributes",
     "pagedata",
     "places",
-    "pocket",
     "preferences",
     "privatebrowsing",
     "prompts",
